framework/src/loops/ComputeLinearFVFaceThread.C

```diff
@@ -76,48 +83,103 @@ ComputeLinearFVFaceThread::join(const ComputeLinearFVFaceThread & /*y*/)
 }
 
 void
-ComputeLinearFVFaceThread::fetchSystemContributionObjects()
+ComputeLinearFVFaceThread::setupSystemContributionObjects()
 {
-  const auto system_number = _fe_problem.getLinearSystem(_linear_system_number).number();
+  // The reason why we need to grab vectors and matrices separately is that
+  // we want to grab a union instead of an intersection.
+  std::vector<LinearFVFluxKernel *> kernels_after_vectors;
+  _fe_problem.theWarehouse()
+      .query()
+      .template condition<AttribSysNum>(_system_number)
+      .template condition<AttribSystem>("LinearFVFluxKernel")
+      .template condition<AttribThread>(_tid)
+      .template condition<AttribVectorTags>(_vector_tags)
+      .queryInto(kernels_after_vectors);
+
+  std::vector<LinearFVFluxKernel *> kernels_after_matrices;
+  _fe_problem.theWarehouse()
+      .query()
+      .template condition<AttribSysNum>(_system_number)
+      .template condition<AttribSystem>("LinearFVFluxKernel")
+      .template condition<AttribThread>(_tid)
+      .template condition<AttribMatrixTags>(_matrix_tags)
+      .queryInto(kernels_after_matrices);
```

Comment:
I wonder if we can improve on this btw. Seems like a "union" would not be something radical to consider in the query system

probably does not matter, unless you see it in profiling